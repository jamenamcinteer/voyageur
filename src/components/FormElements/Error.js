import React from "react";
import styled from "styled-components";

const StyledError = styled.div`
  color: ${props => props.theme.themeColorRed};

  &:first-child {
    margin-top: -10px;
  }
`;

const Error = props => {
  return (
    <React.Fragment>
      {props.errors.filter(err => err.field === props.field).length > 0 &&
        props.errors.map((err, index) => {
          if (err.field === props.field) {
            return (
              <StyledError key={index} style={props.style}>
                {err.error}
              </StyledError>
            );
          }
          return true;
        })}
    </React.Fragment>
  );
};

export default Error;
