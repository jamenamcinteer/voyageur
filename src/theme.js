const theme = {
  themeColor: "#1aae9f",
  themeColorLight: "#a3ded8",
  themeColorRed: "#d85b6e",
  themeColorSecondary: "#c7d2db",
  themeColorSecondaryLight: "#dfe7ed",
  darkFont: "#4b5c6b",
  lightFont: "#788896"
};

export default theme;
