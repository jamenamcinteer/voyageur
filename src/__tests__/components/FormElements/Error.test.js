import React from "react";
import { render, cleanup } from "@testing-library/react";
import Error from "../../../components/FormElements/Error";
import "jest-styled-components";
import theme from "../../../theme";
import { ThemeProvider } from "styled-components";

export function renderWithTheme(component) {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
}

afterEach(cleanup);

test("should render with theme and passed props", () => {
  const errors = [
    {
      field: "testField",
      error: "This field is required."
    },
    {
      field: "testField2",
      error: "This field must be unique."
    }
  ];
  const { asFragment } = renderWithTheme(
    <Error errors={errors} field="testField" />
  );
  expect(asFragment()).toMatchSnapshot();
});
