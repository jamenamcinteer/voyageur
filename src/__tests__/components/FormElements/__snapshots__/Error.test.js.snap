// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with theme and passed props 1`] = `
<DocumentFragment>
  .c0 {
  color: #D24157;
}

.c0:first-child {
  margin-top: -10px;
}

<div
    class="c0"
  >
    This field is required.
  </div>
</DocumentFragment>
`;
